beacon-chain/rpc/beaconv1/blocks.go

```diff
@@ -141,13 +161,13 @@ func (bs *Server) SubmitBlock(ctx context.Context, req *ethpb.BeaconBlockContain
 	blk := req.Message
 	rBlock, err := migration.V1ToV1Alpha1Block(&ethpb.SignedBeaconBlock{Block: blk, Signature: req.Signature})
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not convert block to v1")
+		return nil, status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block")
 	}
 	v1alpha1Block := interfaces.WrappedPhase0SignedBeaconBlock(rBlock)
 
 	root, err := blk.HashTreeRoot()
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not tree hash block: %v", err)
+		return nil, status.Errorf(codes.InvalidArgument, "Could not tree hash block: %v", err)
```

Comment:
This feels more `Internal` than `InvalidArgument` to me

Reply:
Not sure about this one and the other one. `InvalidArgument` will get converted to HTTP status code `400` and https://ethereum.github.io/eth2.0-APIs/#/Beacon/publishBlock says that we should return `400` when
> The SignedBeaconBlock object is invalid

I guess the question is are these errors equivalent to submitting an invalid block?

Reply:
I could see them being both. The error could happen due to 1
1.) Bad input block
2.) Bad HTR implementation 

Although you did use `Internal` for `MarshalSSZ` and they are similar. Perhaps, we can settle into one

![Screen Shot 2021-06-14 at 9 07 34 AM](https://user-images.githubusercontent.com/21316537/121923616-f7b4bf80-ccef-11eb-98ba-fcd627d1e7e1.png)

Reply:
Nvm. That block is from the internal DB. That makes sense to me